src/eth/evm/revm.rs

```diff
@@ -102,7 +116,19 @@ impl Evm for Revm {
         match evm_result {
             Ok(result) => {
                 let session = evm.take_db();
-                Ok(parse_revm_execution(result, session.input, session.storage_changes)?)
+
+                // cache new bytecodes generated during this transaction
+                let has_new_bytecodes = session.has_new_bytecodes();
+                let result = parse_revm_execution(result, session.input, session.storage_changes)?;
+                if result.is_success() && has_new_bytecodes {
+                    let mut bytecodes_write_lock = self.bytecodes.write().unwrap();
+                    for (bytecode_hash, bytecode) in session.new_analysed_bytecodes {
```

Comment:
You can use `HashMap::reserve` to allocate `session.new_analysed_bytecodes.len()` elements before beggining this loop.